lib/docs/token-response-types.md

```diff
@@ -0,0 +1,74 @@
+# Token Response Types
+
+When acquiring tokens, `msal@1.x.x` considers three different response types:
+
+| Response Type | Definition | Reference | Request Example |
+| ------- | -------- | -------- | -------- |
+| **token** | Indicates a request will be made for an Access Token | [OAuth 2.0 Implicit Flow Specification](https://tools.ietf.org/html/rfc6749#section-4.2.1) | [Request example](https://tools.ietf.org/html/rfc6749#section-4.2.1)
+| **id_token** | Indicates a request will be made for an ID Token | [OpenID Connect Specification](https://openid.net/specs/openid-connect-core-1_0.html#Authentication) | [Request Example](https://openid.net/specs/openid-connect-core-1_0.html#id_tokenExample)
+| **id_token token** | Indicates that both an ID Token and Access Token will be requested | [OpenID Connect Specification](https://openid.net/specs/openid-connect-core-1_0.h:tml#Authentication) | [Request Example](https://openid.net/specs/openid-connect-core-1_0.html#id_token-tokenExample)
+
+The table below describes the mapping of `msal@1.x.x` APIs in combination with the user-provided `scopes` to the resulting `response_type` set in the server request.
+
+| Scenarios | Client ID as Only Scope | No Scopes | Access Token Scopes with No client ID | Access Token Scopes with client ID
```

Comment:
Instead of singling out Client ID here can we define what a "Login Scope" is above and then replace any reference to Client ID with "Login Scopes"?

Reply:
Good call

Reply:
@tnorling done, I did some major changes to try to make the concepts more clear and better organized, making sure this equivalence was emphasized. Hopefully the way I wrote it won't add to the confusion.